File: PROGRAMACAO-DE-APLICATIVOS/Sistema de Gerenciamento/models/pessoa.py
class Pessoa:
    """Classe base que representa uma pessoa no sistema."""

    def __init__(self, nome: str, telefone: str, email: str):
        self.__nome = nome
        self.__telefone = telefone
        self.__email = email
        self.__id = None

    # --- Getters e Setters ---

    @property
    def id(self):
        return self.__id

    @id.setter
    def id(self, valor: int):
        if valor is not None and valor <= 0:
            raise ValueError("ID deve ser um número positivo.")
        self.__id = valor

    @property
    def nome(self):
        return self.__nome

    @nome.setter
    def nome(self, valor: str):
        if not valor or not valor.strip():
            raise ValueError("Nome não pode ser vazio.")
        self.__nome = valor.strip()

    @property
    def telefone(self):
        return self.__telefone

    @telefone.setter
    def telefone(self, valor: str):
        self.__telefone = valor

    @property
    def email(self):
        return self.__email

    @email.setter
    def email(self, valor: str):
        self.__email = valor

    def apresentar(self) -> str:
        """Método polimórfico — cada subclasse sobrescreve com sua apresentação."""
        return f"Pessoa: {self.__nome} | Tel: {self.__telefone}"

    def __repr__(self):
        return self.apresentar()

File: PROGRAMACAO-DE-APLICATIVOS/Sistema de Gerenciamento/src/main/java/models/Pessoa.java
package models;

/**
 * Classe base que representa qualquer pessoa no sistema.
 * Aplica encapsulamento com atributos privados e getters/setters.
 */
public abstract class Pessoa {

    private int id;
    private String nome;
    private String telefone;
    private String email;

    public Pessoa(String nome, String telefone, String email) {
        this.setNome(nome);
        this.setTelefone(telefone);
        this.setEmail(email);
    }

    // ── Getters e Setters ──────────────────────────────────────────

    public int getId() {
        return id;
    }

    public void setId(int id) {
        if (id <= 0) throw new IllegalArgumentException("ID deve ser positivo.");
        this.id = id;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        if (nome == null || nome.isBlank())
            throw new IllegalArgumentException("Nome não pode ser vazio.");
        this.nome = nome.trim();
    }

    public String getTelefone() {
        return telefone;
    }

    public void setTelefone(String telefone) {
        this.telefone = telefone;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    // ── Método polimórfico ─────────────────────────────────────────

    /**
     * Cada subclasse sobrescreve este método para se apresentar
     * de forma específica ao seu papel no sistema (polimorfismo).
     */
    public abstract String apresentar();

    @Override
    public String toString() {
        return apresentar();
    }
}
